script/color-contrast.config.ts

```diff
@@ -0,0 +1,225 @@
+    [7, 'fg.default', 'sponsors.muted'],
+    // role
+    [7, 'accent.fg', 'canvas.default'],
+    [7, 'accent.fg', 'canvas.subtle'],
+    [7, 'accent.fg', 'canvas.inset'],
+    [7, 'success.fg', 'canvas.default'],
+    [7, 'success.fg', 'canvas.subtle'],
+    [7, 'success.fg', 'canvas.inset'],
+    [7, 'open.fg', 'canvas.default'],
+    [7, 'open.fg', 'canvas.subtle'],
+    [7, 'open.fg', 'canvas.inset'],
+    [7, 'danger.fg', 'canvas.default'],
+    [7, 'danger.fg', 'canvas.subtle'],
+    [7, 'danger.fg', 'canvas.inset'],
+    [7, 'closed.fg', 'canvas.default'],
+    [7, 'closed.fg', 'canvas.subtle'],
+    [7, 'closed.fg', 'canvas.inset'],
+    [7, 'attention.fg', 'canvas.default'],
+    [7, 'attention.fg', 'canvas.subtle'],
+    [7, 'attention.fg', 'canvas.inset'],
+    [7, 'severe.fg', 'canvas.default'],
+    [7, 'severe.fg', 'canvas.subtle'],
+    [7, 'severe.fg', 'canvas.inset'],
+    [7, 'done.fg', 'canvas.default'],
+    [7, 'done.fg', 'canvas.subtle'],
+    [7, 'done.fg', 'canvas.inset'],
+    [7, 'sponsors.fg', 'canvas.default'],
+    [7, 'sponsors.fg', 'canvas.subtle'],
+    [7, 'sponsors.fg', 'canvas.inset'],
+    // role text on role bg
+    [7, 'accent.fg', 'accent.subtle'],
+    [7, 'success.fg', 'success.subtle'],
+    [7, 'open.fg', 'open.subtle'],
+    [7, 'danger.fg', 'danger.subtle'],
+    [7, 'closed.fg', 'closed.subtle'],
+    [7, 'attention.fg', 'attention.subtle'],
+    [7, 'severe.fg', 'severe.subtle'],
+    [7, 'done.fg', 'done.subtle'],
+    [7, 'sponsors.fg', 'sponsors.subtle'],
+    //
+    [7, 'accent.fg', 'accent.muted'],
+    [7, 'success.fg', 'success.muted'],
+    [7, 'open.fg', 'open.muted'],
+    [7, 'danger.fg', 'danger.muted'],
+    [7, 'closed.fg', 'closed.muted'],
+    [7, 'attention.fg', 'attention.muted'],
+    [7, 'severe.fg', 'severe.muted'],
+    [7, 'done.fg', 'done.muted'],
+    [7, 'sponsors.fg', 'sponsors.muted'],
+    // fg.onEmphasis
+    [7, 'fg.onEmphasis', 'neutral.emphasis'],
+    [7, 'fg.onEmphasis', 'neutral.emphasisPlus'],
+    [7, 'fg.onEmphasis', 'accent.emphasis'],
+    [7, 'fg.onEmphasis', 'success.emphasis'],
+    [7, 'fg.onEmphasis', 'open.emphasis'],
+    [7, 'fg.onEmphasis', 'danger.emphasis'],
+    [7, 'fg.onEmphasis', 'closed.emphasis'],
+    [7, 'fg.onEmphasis', 'attention.emphasis'],
+    [7, 'fg.onEmphasis', 'severe.emphasis'],
+    [7, 'fg.onEmphasis', 'done.emphasis'],
+    [7, 'fg.onEmphasis', 'sponsors.emphasis'],
+  ] as ContrastRequirement[]
+]
+
+export const cavnasColors: string[] = ['canvas.default', 'canvas.subtle']
+
+export const contrastRequirements: { [key: string]: ContrastRequirement[] } = {
+  // default light mode
+  light: baseRequirements,
+  light_high_contrast: highContrast,
+  light_colorblind: baseRequirements,
+  light_tritanopia: baseRequirements,
+  // default dark mode
+  dark: baseRequirements,
+  dark_dimmed: baseRequirements,
+  dark_high_contrast: highContrast,
+  dark_colorblind: baseRequirements,
+  dark_tritanopia: baseRequirements,
+}
```

Comment:
Question around maintainability of this file in future..

Is it worth co-locating this information to the the functional src tokens as additional metadata per token? If the plan is to add _more_ functional tokens in future to Primitives, it can be easy to forget to update this file. We could also do something interesting by using style dictionary custom JSON format  to generate the accessibility lookup tables.

Haven't thought this through entirely yet, but the concern is coming from risks to long-term maintainability of this file. That said, if I'm wrong on my assumption that we add functional tokens regularly, this is a moot point.

Reply:
I would love this. Adding a metadata part on the tokens sounds pretty good. Than we can do something like with `deprecated`. It could also be used for docs to show contrast ratios, etc. if we want. And to suggest pairing in `vscode` if this is possible.

Reply:
But this would be after the PR is merged. I'll create an issue for it.